client/src/test/java/io/pravega/client/stream/mock/MockClientFactory.java

```diff
@@ -92,6 +94,11 @@ public MockClientFactory(String scope, Controller controller, ConnectionPool con
         return impl.createRevisionedStreamClient(streamName, serializer, config);
     }
 
+    @Override
+    public <T> RevisionedStreamClient<T> createRevisionedStreamClient(String streamName, Controller controllerObj, Serializer<T> serializer, SynchronizerConfig config, AccessOperation accessOperation) {
+        return impl.createRevisionedStreamClient(streamName, serializer, config);
```

Comment:
this method is ignoring the passed controller obj.. i know this is mocked, but this can impact the expectations of a test written with this where they may want to mock the controller responses.

Reply:
Fixed.